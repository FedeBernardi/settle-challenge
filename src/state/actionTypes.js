export const DEPOSIT_FIAT = 'DEPOSIT_FIAT';
