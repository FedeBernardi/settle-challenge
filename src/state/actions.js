import * as actionTypes from './actionTypes';

export const depositFiat = amount => ({
  type: actionTypes.DEPOSIT_FIAT,
  payload: {
    amount
  }
})
