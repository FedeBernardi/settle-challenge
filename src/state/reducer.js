import * as actionTypes from './actionTypes';

const reducer = (state, { type, payload }) => {
  switch(type) {
    case actionTypes.DEPOSIT_FIAT:
      state.wallet.ars = state.wallet.ars + payload.amount;

      return JSON.parse(JSON.stringify(state));
    default:
      return state;
  }
}

export default reducer;
