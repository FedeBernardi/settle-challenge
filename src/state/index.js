import React, { createContext, useReducer } from 'react';

import { CURRENCIES } from '../constants';
import reducer from './reducer';

const initialState = {
  currenciesPrice: {
    [CURRENCIES.BTC]: 0,
    [CURRENCIES.ETH]: 0,
    [CURRENCIES.USDC]: 0
  },
  wallet: {
    [CURRENCIES.BTC]: 0,
    [CURRENCIES.ETH]: 0,
    [CURRENCIES.USDC]: 0,
    [CURRENCIES.ARS]: 0
  },
  fees: 0,
  ordersHistory: []
}

export const AppContext = createContext(null);

export const AppContextProvider = ({ children }) => (
  <AppContext.Provider value={useReducer(reducer, initialState)}>
    {children}
  </AppContext.Provider>
) 
