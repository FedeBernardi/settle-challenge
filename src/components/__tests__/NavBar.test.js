import { render, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { AppContextProvider } from '../../state';

import NavBar from '../NavBar';

describe('NavBar Component', () => {
  const renderComponent = () => render(
    <AppContextProvider>
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>
    </AppContextProvider>
  );
  
  it('matches snapshot', () => {
    const { container } = renderComponent();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('redirects to /trade when clicking the Trade link', () => {
    const { getByText } = renderComponent();

    const ordersButton = getByText(/trade/i);
    fireEvent.click(ordersButton);

    expect(window.location.pathname).toBe('/trade');
  });

  it('redirects to /wallet when clicking the Wallet link', () => {
    const { getByText } = renderComponent();

    const ordersButton = getByText(/wallet/i);
    fireEvent.click(ordersButton);

    expect(window.location.pathname).toBe('/wallet');
  });

  it('redirects to /deposit when clicking the Deposit link', () => {
    const { getByText } = renderComponent();

    const ordersButton = getByText(/deposit/i);
    fireEvent.click(ordersButton);

    expect(window.location.pathname).toBe('/deposit');
  });
});
