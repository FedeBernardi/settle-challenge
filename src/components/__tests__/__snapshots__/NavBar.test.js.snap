// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component matches snapshot 1`] = `
<nav
  class="sc-bdvvtL eluuLJ"
>
  <a
    href="/trade"
  >
    <span
      class="sc-gsDKAQ eqyydY"
    >
      Trade
    </span>
  </a>
  <a
    href="/wallet"
  >
    <span
      class="sc-gsDKAQ eqyydY"
    >
      Wallet
    </span>
  </a>
  <a
    href="/deposit"
  >
    <span
      class="sc-gsDKAQ eqyydY"
    >
      Deposit
    </span>
  </a>
  <span
    class="fees-counter"
  >
    Fees: 0.00 ARS
  </span>
</nav>
`;
